R/plot_carp.R

```diff
@@ -32,38 +32,39 @@
 #'          clusters is used.
+#' @param percent.seq A grid of values of percent along which to generate dynamic
+#'                    visualizations (if dynamic == TRUE)
 #' @param ... Additional arguments, which are handled differently for different
 #'            values of \code{type}.\itemize{
-#'            \item When \code{type} is \code{"path"}, \code{"dynamic_path"},
-#'                  or \code{"interactive"}, the presence of
+#'            \item When \code{type} is \code{"path"}, the presence of
 #'                  unknown arguments triggers an error;
 #'            \item when \code{type == "dendrogram"}
 #'                  \code{...} is forwarded to \code{\link[stats]{dendrogram}}; and
-#'            \item when \code{type == "heatmap"}, \code{...} is forwarded to
-#'                  \code{\link[heatmaply]{heatmaply}}.
-#'            } See the documentation of the linked functions for details about
-#'            additional supported arguments. \code{saveviz} passes arguments
-#'            to the corresponding plot \code{type}.
-#' @param dend.branch.width Branch width for dendrogram plots (ignored for
-#'        other plot types) - must be positive.
-#' @param dend.labels.cex Label size for dendrogram plots (ignored for other plot
-#'        types) - must be positive.
-#' @param dend.ylab.cex Y-axis size for dendrogram plots (ignored for other plot
-#'        types) - must be positive.
+#'            \item when \code{type == "heatmap"} and \code{interactive == TRUE},
+#'                  \code{...} is forwarded to \code{\link[heatmaply]{heatmaply}}.
+#'            } See the documentation of \code{\link[ggplot2]{ggplot}} (for \code{interactive == FALSE})
+#'            or \code{\link[plotly]{plotly}} (for \code{interactive == TRUE}) for details about additional
+#'            supported arguments to the corresponding plot \code{type}.
 #' @param slider_y A number to adjust the slider's vertical position for
 #'                 interactive dendrogram and interactive heatmap plots
 #'                 (ignored for other plot types).
-#' @return The value of the return type depends on the \code{type} argument:\itemize{
-#'   \item if \code{type = "dendrogram"}, \code{x} is returned invisibly;
-#'   \item if \code{type = "path"}, an object of class \code{\link[ggplot2]{ggplot}}
+#' @param refit A logical scalar. Should "naive" centroids (TRUE) or the
+#'              actual centroids estimated by convex clustering be used?
+#'              When the default refit = FALSE, the estimated U from the convex
+#'              clustering problem is used. The refit = TRUE returns actual
+#'              centroids (mean) of all elements assigned to that cluster;
+#'              Due to the global shrinkage imposed, these clusters are
+#'              more "shrunk together" than the naive clusters. Only for the
+#'              heatmap plots. (ignored for other plot types).
```

Comment:
Two questions here: 

1) Can we support `refit` in all plot types? I would think it could be useful everywhere. 

2) Why did we decide the default should be `TRUE`?

Reply:
1. The `refit` argument is only supported  in `get_clustered_data` to provide the heatmaps with smooth transmission. Thus I think it only can support the heatmaps.

2. The default of `refit` is `FALSE`, which aims to provide smoother plots automatically.

Reply:
Where else do you need `refit` for path plots? I think `get_cluster_centroids` has it as well (https://github.com/DataSlingers/clustRviz/blob/master/R/accessors_carp.R#L130)

Reply:
I used `get_feature_paths` for path plots so that it can produce values after PC.

Reply:
Ah - I see. That's probably something that should be fixed. I'll open an issue for it.